lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1097,12 +930,19 @@ def reset_stat(tarinfo):
                         continue
 
                 # Dealing with a normal file, just add it by name.
-                tar_file.add(os.path.realpath(b_src_path), arcname=filename, recursive=False, filter=reset_stat)
+                tar_file.add(
+                    to_native(os.path.realpath(b_src_path)),
+                    arcname=filename,
+                    recursive=False,
+                    filter=reset_stat,
+                )
 
-        shutil.copy(b_tar_filepath, b_tar_path)
+        shutil.copy(to_native(b_tar_filepath), to_native(b_tar_path))
```

Comment:
Same here, don't we usually use byte strings for file operations.

Reply:
Same, it accepts `StrPath`: https://github.com/python/typeshed/blob/869238e587ce2bb2e1b990026d5a038f6eb0f1a3/stdlib/2and3/shutil.pyi#L55-L62